x-pack/test/functional/services/logs_ui/log_stream.ts

```diff
@@ -13,8 +14,8 @@ export function LogStreamPageProvider({ getPageObjects, getService }: FtrProvide
   const testSubjects = getService('testSubjects');
 
   return {
-    async navigateTo() {
-      pageObjects.infraLogs.navigateToTab('stream');
+    async navigateTo(params?: Record<string, RisonValue>) {
```

Comment:
Given that the logs stream url only supports a limited set of parameters, how about defining a narrower type that enables auto-completion to help the test authors?

Reply:
Maybe I can reuse the `interface`s from the `With*UrlState` components. I'll give it a try.

Reply:
This is the cleanest I could come up with.

```ts
import { LogPositionUrlState } from '../../../../x-pack/plugins/infra/public/containers/logs/log_position/with_log_position_url_state';

// This can be completed as we go
export interface TabsParams {
  stream: {
    logPosition?: Partial<LogPositionUrlState>;
  };
  settings: never;
  'log-categories': never;
  'log-rate': never;
}

export function InfraLogsPageProvider({ getPageObjects, getService }: FtrProviderContext) {
  // ...
  
    async navigateToTab<T extends LogsUiTab>(logsUiTab: T, params?: TabsParams[T]) {
      let queryString = '';
      if (params) {
        const qsPairs = [];
        // Note the change from `Object.keys()` here. The return type of that function is always `string[]`,
		// Using `for..in` keeps the types.

        for (const key in params) {
          if (params.hasOwnProperty(key)) {
			// There's no way around this type cast.
            const value = (params[key] as unknown) as RisonValue;
            qsPairs.push(`${key}=${encode(value)}`);
          }
        }
        queryString = '?' + qsPairs.join('&');
      }
		
      // ...
    },
  // ...
}
```